doc/_extensions/zephyr/api_overview.py

```diff
@@ -0,0 +1,161 @@
+# Copyright (c) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import doxmlparser
+
+from docutils import nodes
+from doxmlparser.compound import DoxCompoundKind
+from pathlib import Path
+from sphinx.application import Sphinx
+from sphinx.util.docutils import SphinxDirective
+from typing import Any, Dict
+
+
+class ApiOverview(SphinxDirective):
+    r"""
+    This is a Zephyr directive to generate a table containing an overview
+    of all APIs. This table will show the API name, version and since which
+    version it is present - all information extracted from Doxygen XML output.
+
+    It is exclusively used by the doc/develop/api/overview.rst page.
+
+    Configuration options:
+    api_overview_doxygen_xml_dir: Doxygen xml output directory
+    api_overview_doxygen_base_url: Doxygen base html directory
+    """
+
+    def run(self):
+        return [self.env.api_overview_table]
+
+
+spacer = '<span style="width:{indent}px;display:inline-block">&nbsp;</span>'
```

Comment:
looks hacky, and HTML only (important as some still insist on PDF)

Reply:
+1, please stick to standard nodes and/or rely on CSS for tweaking the look&feel.

Reply:
Ok, will try to convey the hierarchy with `-` instead.